source/PrintData.cpp

```diff
@@ -0,0 +1,497 @@
+		// Maximum heat is 100 degrees per ton. Bleed off rate is 1/1000 per 60th of a second, so:
+		cout << 60. * ship.HeatDissipation() * ship.MaximumHeat() << ',';
+
+		int numTurrets = 0;
+		int numGuns = 0;
+		for(auto &hardpoint : ship.Weapons())
+		{
+			if(hardpoint.IsTurret())
+				++numTurrets;
+			else
+				++numGuns;
+		}
+		cout << numGuns << ',' << numTurrets << ',';
+
+		int numFighters = ship.BaysTotal("Fighter");
+		int numDrones = ship.BaysTotal("Drone");
+		cout << numFighters << ',' << numDrones << ',';
+
+		double deterrence = 0.;
+		for(const Hardpoint &hardpoint : ship.Weapons())
+			if(hardpoint.GetOutfit())
+			{
+				const Outfit *weapon = hardpoint.GetOutfit();
+				if(weapon->Ammo() && !ship.OutfitCount(weapon->Ammo()))
+					continue;
+				double damage = weapon->ShieldDamage() + weapon->HullDamage()
+					+ (weapon->RelativeShieldDamage() * ship.Attributes().Get("shields"))
+					+ (weapon->RelativeHullDamage() * ship.Attributes().Get("hull"));
+				deterrence += .12 * damage / weapon->Reload();
+			}
+		cout << deterrence << '\n';
+	}
+}
+
+
+
+void PrintData::PrintWeaponStats()
+{
+	cout << "name" << ',' << "category" << ',' << "cost" << ',' << "space" << ',' << "range" << ','
+		<< "reload" << ',' << "burst count" << ',' << "burst reload" << ',' << "lifetime" << ','
+		<< "shots/second" << ',' << "energy/shot" << ',' << "heat/shot" << ',' << "recoil/shot" << ','
+		<< "energy/s" << ',' << "heat/s" << ',' << "recoil/s" << ',' << "shield/s" << ','
+		<< "discharge/s" << ',' << "hull/s" << ',' << "corrosion/s" << ',' << "heat dmg/s" << ','
+		<< "burn dmg/s" << ',' << "energy dmg/s" << ',' << "ion dmg/s" << ',' << "slow dmg/s" << ','
+		<< "disruption dmg/s" << ',' << "piercing" << ',' << "fuel dmg/s" << ',' << "leak dmg/s" << ','
+		<< "push/s" << ',' << "homing" << ',' << "strength" << ',' << "deterrence" << '\n';
+
+	for(auto &it : GameData::Outfits())
+	{
+		// Skip non-weapons and submunitions.
+		if(!it.second.IsWeapon() || it.second.Category().empty())
+			continue;
+
+		const Outfit &outfit = it.second;
+		cout << it.first << ',';
+		cout << outfit.Category() << ',';
+		cout << outfit.Cost() << ',';
+		cout << -outfit.Get("weapon capacity") << ',';
+
+		cout << outfit.Range() << ',';
+
+		double reload = outfit.Reload();
+		cout << reload << ',';
+		cout << outfit.BurstCount() << ',';
+		cout << outfit.BurstReload() << ',';
+		cout << outfit.TotalLifetime() << ',';
+		double fireRate = 60. / reload;
+		cout << fireRate << ',';
+
+		double firingEnergy = outfit.FiringEnergy();
+		cout << firingEnergy << ',';
+		firingEnergy *= fireRate;
+		double firingHeat = outfit.FiringHeat();
+		cout << firingHeat << ',';
+		firingHeat *= fireRate;
+		double firingForce = outfit.FiringForce();
+		cout << firingForce << ',';
+		firingForce *= fireRate;
+
+		cout << firingEnergy << ',';
+		cout << firingHeat << ',';
+		cout << firingForce << ',';
+
+		double shieldDmg = outfit.ShieldDamage() * fireRate;
```

Comment:
all these variables for clarity's sake?
If this is how we're going to do it, do the same thing in PrintEnginesStat, using small variables for clarity?
Or do you just do it because these are longer? Some in printLoadedStats are as big but you didnt use a variable.

Basically what I'm saying is follow the same logic everywhere for defining variables or not.

Reply:
This part was just copy-pasted. It does keep the lines short, though.